src/Ztm.WebApi/Startup.cs

```diff
@@ -86,7 +90,8 @@ IZcoinRpcClientFactory CreateZcoinRpcClientFactory(IServiceProvider provider)
             return new ZcoinRpcClientFactory(
                 config.Rpc.Address,
                 config.Network.Type,
-                RPCCredentialString.Parse($"{config.Rpc.UserName}:{config.Rpc.Password}")
+                RPCCredentialString.Parse($"{config.Rpc.UserName}:{config.Rpc.Password}"),
+                (ITransactionEncoder)provider.GetRequiredService(typeof(ITransactionEncoder))
```

Comment:
We can use `provider.GetRequiredService<ITransactionEncoder>()` to remove casting.